sdk/eventgrid/azure-messaging-eventgrid-namespaces/src/main/java/com/azure/messaging/eventgrid/namespaces/EventGridReceiverAsyncClient.java

```diff
@@ -385,12 +387,48 @@ public Mono<ReceiveResult> receive(Integer maxEvents, Duration maxWaitTime) {
             requestOptions.addQueryParam("maxEvents", String.valueOf(maxEvents), false);
         }
         if (maxWaitTime != null) {
+            // Ensure the http client timeout is longer than the maxWaitTime
+            addTimeoutToContext(requestOptions, maxWaitTime);
             requestOptions.addQueryParam("maxWaitTime", String.valueOf(maxWaitTime.getSeconds()), false);
         }
         return receiveWithResponse(topicName, subscriptionName, requestOptions).flatMap(FluxUtil::toMono)
             .map(protocolMethodData -> protocolMethodData.toObject(ReceiveResult.class));
     }
 
+    /**
+     * Receive a batch of Cloud Events from a subscription.
+     *
+     * @param maxEvents Max Events count to be received. Minimum value is 1, while maximum value is 100 events. If not
+     * specified, the default value is 1.
+     * @param maxWaitTime Max wait time value for receive operation in Seconds. It is the time in seconds that the
+     * server approximately waits for the availability of an event and responds to the request. If an event is
+     * available, the broker responds immediately to the client. Minimum value is 10 seconds, while maximum value is 120
+     * seconds. If not specified, the default value is 60 seconds.
```

Comment:
So, if `maxWaitTime` is `null` the service waits for 60 seconds by default?

Reply:
Correct. Now that you've asked I should probably set the HTTP timeout in the unspecified case to 65 seconds.